app/main/views/dashboard.py

```diff
@@ -47,31 +48,38 @@ def service_dashboard(service_id):
     if not current_user.has_permissions("view_activity"):
         return redirect(url_for("main.choose_template", service_id=service_id))
 
-    notifications = notification_api_client.get_notifications_for_service(
-        service_id=service_id,
-    )["notifications"]
+    job_response = job_api_client.get_jobs(service_id)["data"]
+    notifications_response = notification_api_client.get_notifications_for_service(service_id)["notifications"]
+    service_data_retention_days = 7
 
-    job_response = job_api_client.get_jobs(service_id)
+    aggregate_notifications_by_job = defaultdict(list)
+    for notification in notifications_response:
+        job_id = notification.get("job", {}).get("id")
+        if job_id:
+            aggregate_notifications_by_job[job_id].append(notification)
 
-    service_data_retention_days = 7
-    jobs = [
+    job_and_notifications = [
         {
             "job_id": job["id"],
             "time_left": get_time_left(job["created_at"]),
             "download_link": url_for(
                 ".view_job_csv", service_id=current_service.id, job_id=job["id"]
             ),
+            "view_job_link": url_for(
+                ".view_job", service_id=current_service.id, job_id=job["id"]
+            ),
+            "created_at": job["created_at"],
             "notification_count": job["notification_count"],
             "created_by": job["created_by"],
+            "notifications": aggregate_notifications_by_job.get(job["id"], []),
```

Comment:
Here, I'm adding the aggregate notifications in a list based on job id

Reply:
Excellent!